app/models/heart_rate_data.rb

```diff
@@ -0,0 +1,30 @@
+class HeartRateData < ApplicationRecord
+
+  belongs_to :user, dependent: :destroy
+
+  def update_heart_rate_data!
+    update!(data: heart_rate_data_for_the_day)
+  end
+
+  private
+
+  def heart_rate_data_for_the_day
+    client.heart_rate_intraday_time_series(
+      user_id: '-',
+      date: 'today',
+      start_time: '00:00',
+      end_date: '24:00',
```

Comment:
Are you sure the `start_time` and `end_date` values are different? I'm thinking they may be both be interpreted as midnight. Also, `end_date` should be `end_time` if it is actually expressing a time value.